hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieSparkClusteringClient.java

```diff
@@ -53,11 +53,15 @@ public void cluster(HoodieInstant instant) throws IOException {
     SparkRDDWriteClient<T> writeClient = (SparkRDDWriteClient<T>) clusteringClient;
     Option<HoodieCommitMetadata> commitMetadata = writeClient.cluster(instant.getTimestamp(), true).getCommitMetadata();
     Stream<HoodieWriteStat> hoodieWriteStatStream = commitMetadata.get().getPartitionToWriteStats().entrySet().stream().flatMap(e ->
-            e.getValue().stream());
+        e.getValue().stream());
     long errorsCount = hoodieWriteStatStream.mapToLong(HoodieWriteStat::getTotalWriteErrors).sum();
     if (errorsCount > 0) {
       // TODO: Should we treat this fatal and throw exception?
       LOG.error("Clustering for instant (" + instant + ") failed with write errors");
     }
+    if (writeClient != clusteringClient) {
+      // Clustering write client has been updated with new schema, closing the old one
+      writeClient.close();
```

Comment:
guess, we could miss to close the old write client if there is no clustering inflight. i.e. when async clustering thread is just waiting and if write client is updated (due to schema change), we might over ride w/ new write client and we may not close the old one. Can we ensure we close it out in all cases.

Reply:
Right.  I've fixed the logic to consider such cases so that the old write client is properly closed in all cases.